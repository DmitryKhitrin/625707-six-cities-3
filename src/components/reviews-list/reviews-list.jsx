import React from "react";
import {PropTypes} from "prop-types";
import {ReviewsItem} from "../reviews-item/reviews-item.jsx";

export const ReviewsList = ({reviews}) => {
  if (!reviews || reviews.length < 0) {
    return null;
  }
  return (
    <ul className="reviews__list">
      {reviews
          .sort((a, b) => new Date(b.date) - new Date(a.date))
          .slice(0, 10)
          .map(
              ({
                id,
                date,
                comment,
                rating,
                user = {},
              }) => {
                const {personName = ``, personPhoto = ``} = user;
                return (
                  <ReviewsItem
                    key={id}
                    reviewsDate={date}
                    reviewsText={comment}
                    reviewsRating={rating}
                    reviewsUserName={personName}
                    reviewsAvatar={personPhoto}
                  />
                );
              }
          )}
    </ul>
  );
};

ReviewsList.propTypes = {
  reviews: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        date: PropTypes.string,
        comment: PropTypes.string,
        rating: PropTypes.string,
        personName: PropTypes.string,
        personPhoto: PropTypes.string
      })
  )
};
