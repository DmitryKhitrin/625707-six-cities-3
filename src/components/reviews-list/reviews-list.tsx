import React, {FC, memo} from "react";
import {ReviewsItem} from "../reviews-item/reviews-item";
import {ParsedComment} from '../../utils/utils';

type Props = {
  reviews: ParsedComment[];
}

const ReviewsList: FC<Props> = ({reviews}) => {
  if (!reviews || reviews.length < 0) {
    return null;
  }
  return (
    <ul className="reviews__list">
      {reviews
          .sort((a, b) => b.date - a.date)
          .slice(0, 10)
          .map(
              ({
                id,
                date,
                comment,
                rating,
                user = {},
              }) => {
                const {personName = ``, personPhoto = ``} = user;
                return (
                  <ReviewsItem
                    key={id}
                    reviewsDate={date}
                    reviewsText={comment}
                    reviewsRating={rating}
                    reviewsUserName={personName}
                    reviewsAvatar={personPhoto}
                  />
                );
              }
          )}
    </ul>
  );
};

const MemoizedReviewsList = memo(ReviewsList);
export {MemoizedReviewsList as ReviewsList};
