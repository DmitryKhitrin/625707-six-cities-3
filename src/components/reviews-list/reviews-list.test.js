import React from "react";
import renderer from "react-test-renderer";
import {ReviewsList} from "./reviews-list.jsx";

export const mocData = {
  placePhotosList: [],
  offerHeader: `Test header`,
  descriptions: [`This is kind of nice place.`],
  isPremium: true,
  placeType: `Apartment`,
  rating: `90`,
  bedroomsCount: `10`,
  maxPeopleCount: 100,
  price: 500,
  amenitiesList: [],
  reviews: [
    {
      id: `1`,
      reviewsDate: `April 2019`,
      reviewsText: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      reviewsRating: `80%`,
      reviewsUserName: `Max`,
      reviewsAvatar: `img/avatar-max.jpg`
    }
  ],
  offersList: [
    {
      id: `5`,
      price: 80,
      previewImage: `img/apartment-01.jpg`,
      rating: `80%`,
      type: `Apartment`,
      isPremium: true,
      title: `Wood and stone place`,
      location: [52.369553943508, 4.939309666406197]
    },
    {
      id: `3`,
      price: 132,
      previewImage: `img/apartment-02.jpg`,
      rating: `80%`,
      type: `Apartment`,
      isPremium: false,
      title: `Canal View Prinsengracht`,
      location: [52.3909553943508, 4.929309666406198]
    },
    {
      id: `4`,
      price: 180,
      previewImage: `img/apartment-03.jpg`,
      rating: `100%`,
      type: `Apartment`,
      isPremium: true,
      title: `Nice, cozy, warm big bed apartment`,
      location: [52.3809553943508, 4.939309666406198]
    }
  ]
};

it(`<ReviewsList /> should render ReviewsList.`, () => {
  const tree = renderer.create(<ReviewsList {...mocData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
