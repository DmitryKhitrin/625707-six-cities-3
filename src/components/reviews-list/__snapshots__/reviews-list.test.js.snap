// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewsList /> should render ReviewsList. 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height={54}
          src=""
          width={54}
        />
      </div>
      <span
        className="reviews__user-name"
      >
        
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      />
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      />
    </div>
  </li>
</ul>
`;
